### DISCUSSION TITLE
Trying to combine two stores

### DISCUSSION THREAD

**Discussion by FlorianGoussin** [pmndrs/zustand#1854] [General] [answered]
Hello, 
I'm trying to combine two stores and I tried to implement this [solution](https://github.com/pmndrs/zustand/issues/228) which seems to work for [sepehr500](https://github.com/sepehr500) and I ended up with some errors. I also started this [thread](https://stackoverflow.com/questions/76444458/combine-two-stores-with-react-and-zustang) on stackoverflow.
Here is my code:

```javascript
import { create } from 'zustand'
import { temporal } from 'zundo'

const createSelectionStore = create((set) => ({
    selected: [], // selected path indexes
    select: (index) =>
        ...
    deselect: () =>
        ...
}))

const createBikePathStore = create((set) => ({
    paths: [],
    visible: false,
    setPaths: (paths) =>
        ...
    setVisible: (visible) =>
        ...
}))

const useBikePathStore = create((set) => ({
    ...createSelectionStore(set),
    ...createBikePathStore(set),
}))

export default useBikePathStore
```

I get this [error](https://i.stack.imgur.com/dsuFu.png). I first thought I had an issue with **Zundo**, but as you can see I'm not using the `temporal` variable anymore and still have the same error.

Could someone help me with that?

I'm probably missing something so I posted this in general.
https://github.com/pmndrs/zustand/issues/228#issuecomment-1585427367

**Comment by dbritto-dev** [comment, marked as answer]
@FlorianGoussin what you are looking for is the slices pattern https://docs.pmnd.rs/zustand/guides/slices-pattern

The solution 

```ts
const createSelectionStore = (set) => ({ ... })
const createBikePathStore = (set) => ({ ... })
```

> **Note:** you shouldn't use create with `createSelectionStore` and `createBikePathStore` those are slices, just regular functions

**Comment by FlorianGoussin** [comment]
I knew I was missing something. It makes sense that `create` should only be called once. 
Thank you very much! It works now! 👍

**Comment by Klerith** [comment]
What about if you have two slices with the same property or method names?

I'm trying to separate the bound store like this:
```

interface BoundStore { 
  bearSlice: BearSlice;
  birdSlice: BirdSlice & BirdActions;
};
   
  

export const useBoundStore = create<BoundStore>()((...a) => ({
  bearSlice: createBearSlice(...a),
  birdSlice: createBirdSlice(...a),
}))
```

I'm struggling to change the state, however, I was able to get the state with no issues.

**Reply by dai-shi** [reply]
> What about if you have two slices with the same property or method names?

I think what you want is "namespacing".
See #178. It's out of the scope of zustand core library, and the scope of third-party libs.
https://github.com/pmndrs/zustand/blob/2be79c9154685569e635496c3a7257a812ecd0f2/docs/integrations/third-party-libraries.md?plain=1#L18
